napari_aicssegmentation/widgets/collapsible_box.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+
+from qtpy.QtGui import QPixmap
+from qtpy.QtWidgets import (
+    QFrame,
+    QHBoxLayout,
+    QLabel,
+    QVBoxLayout,
+    QWidget
+)
+
+DIR = Path.cwd() / "napari_aicssegmentation"
+
+
+"""
+A collapsible box widget containing a title box and a content box. The title box can be clicked
+to toggle the visibility of the content box.
+
+Params:
+    step:       Number representing the workflow step. This will be displayed as part of the
+                    title and be used as a part of the object name for identification.
+    title:      String
+    content:    QLayout to be nested inside the content box
+    isOpen:     Boolean, whether the widget is open or collapsed
+    isEnabled:  Boolean, overwrites the built-in property QWidget.enabled property
+"""
+
+
+class CollapsibleBox(QWidget):
+    def __init__(self, step, title, content, isOpen=True, isEnabled=True):
+        super().__init__()
+        self.step = step
+        self.title = title
+        self.content = content
+        self.isOpen = isOpen
+        self.setEnabled(isEnabled)
```

Comment:
I think I understand why you're trying to "overwrite" the enabled/disabled functionality of the Widget but it looks like this might not be enough? I'm seeing functionality to set the enabled/disabled state at creation of the widget, but what we really want I believe is to overwrite the `setEnabled` method of the Widget so that we can actually change it's state dynamically.
I want to be able to to things like this:
```
box = CollapsibleBox()
box.setEnabled(False)

def handleSomeEvent():
    box.setEnabled(True)
```
And see the whole widget be enabled / disabled on command. Rather than having to specify it at creation.

Reply:
Oh, that is very true. I don't need to overwrite it at all, I'll just remove this line and the `isEnabled` parameter.